internal/serve/graphql/resolvers/mutations.resolvers.go

```diff
@@ -84,6 +91,110 @@ func (r *mutationResolver) DeregisterAccount(ctx context.Context, input graphql1
 	}, nil
 }
 
+// BuildTransactions is the resolver for the buildTransactions field.
+func (r *mutationResolver) BuildTransactions(ctx context.Context, input graphql1.BuildTransactionsInput) (*graphql1.BuildTransactionsPayload, error) {
+	transaction := input.Transaction
+
+	ops, err := transactionsUtils.BuildOperations(transaction.Operations)
+	if err != nil {
+		return nil, &gqlerror.Error{
+			Message: fmt.Sprintf("Invalid operations: %s", err.Error()),
+			Extensions: map[string]interface{}{
+				"code": "INVALID_OPERATIONS",
+			},
+		}
+	}
+
+	// Convert simulation result if provided
+	var simulationResult entities.RPCSimulateTransactionResult
+	if transaction.SimulationResult != nil {
+		simulationResult = entities.RPCSimulateTransactionResult{
+			Events: transaction.SimulationResult.Events,
+		}
+
+		if transaction.SimulationResult.MinResourceFee != nil {
+			simulationResult.MinResourceFee = *transaction.SimulationResult.MinResourceFee
+		}
+		if transaction.SimulationResult.Error != nil {
+			simulationResult.Error = *transaction.SimulationResult.Error
+		}
+		if transaction.SimulationResult.LatestLedger != nil {
+			simulationResult.LatestLedger = int64(*transaction.SimulationResult.LatestLedger)
+		}
+
+		// Handle TransactionData if provided
+		if transaction.SimulationResult.TransactionData != nil {
+			var txData xdr.SorobanTransactionData
+			if txDataErr := xdr.SafeUnmarshalBase64(*transaction.SimulationResult.TransactionData, &txData); txDataErr != nil {
+				return nil, &gqlerror.Error{
+					Message: fmt.Sprintf("Invalid TransactionData: %s", txDataErr.Error()),
+					Extensions: map[string]interface{}{
+						"code": "INVALID_TRANSACTION_DATA",
+					},
+				}
+			}
+			simulationResult.TransactionData = txData
+		}
+	}
+
+	tx, err := r.transactionService.BuildAndSignTransactionWithChannelAccount(ctx, ops, int64(transaction.Timeout), simulationResult)
+	if err != nil {
+		switch {
+		case errors.Is(err, transactionservices.ErrInvalidArguments):
+			return nil, &gqlerror.Error{
+				Message: err.Error(),
+				Extensions: map[string]interface{}{
+					"code": "INVALID_ARGUMENTS",
+				},
+			}
```

Comment:
Can we have a more specific error message/code here? The client won't know which arguments are invalid.